commitizen/commands/commit.py

```diff
@@ -110,17 +110,20 @@ def __call__(self):
         if dry_run:
             raise DryRunExit()
 
-        signoff: bool = (
-            self.arguments.get("signoff") or self.config.settings["always_signoff"]
-        )
+        always_signoff: bool = self.config.settings["always_signoff"]
+        signoff: bool = self.arguments.get("signoff")
+
+        extra_args = self.arguments.get("extra_cli_args", "")
 
         if signoff:
             out.warn(
                 "signoff mechanic is deprecated, please use `cz commit -- -s` instead."
             )
-            extra_args = self.arguments.get("extra_cli_args", "--") + " -s"
-        else:
-            extra_args = self.arguments.get("extra_cli_args", "")
+
+        if always_signoff or signoff:
+            if extra_args:
+                extra_args += " "
+            extra_args += "-s"
```

Comment:
Will there be any issue if we just do 
```suggestion
            extra_args = f"{extra_args} -s"
```

an additional space should be fine I think 🤔  or we can do `strip()` afterward.

Reply:
I do not like working "abusing" the arguments parsers, especially knowing we should eventually move to `shell=False` everywhere and rewrite all `subprocess.run` calls into `args=['git', 'commit', ...]` arrays to better control the executions.

Reply:
What about

```python
            extra_args = f"{extra_args} -s".strip()
```

Reply:
:+1: